libs/dal/src/repositories/notification/notification.entity.ts

```diff
@@ -38,6 +38,7 @@ export class NotificationEntity {
   updatedAt?: string;
   expireAt?: string;
   tags?: string[];
+  preferences?: WorkflowChannelPreferences;
```

Comment:
This is probably the most uncomfortable part - `preferences` isn't stored on the `NotificationTemplateEntity` for _stateful_ executions, this change here is made to keep in line with mapping of `tags` for stateless workflow executions to the `NotificationTemplateEntity` which the SubscriberJobBound use-case deals with via injection.

Reply:
Does this change mean that we will start persisting them in Mongo on the Notification entity? If not mistaken, we should do that as it's the part that we actually copy the workflow & preference configuration from its blueprint. 

Am I missing something?

Reply:
No, it doesn't mean we will start persisting them. This is purely to make Typescript happy and avoid a sweeping change to the way NotificationTemplate settings are injected in the send-message flow.

Workflow Preferences are stored on the `Preferences` entity instead of the Workflow with this release, to provide a single source of truth for all preference lookups. The downside of this approach is that without making a sweeping change to the send-message flow, this property sits as an artifact on the NotificaitonTemplateEntity until we do a refactor.

Reply:
One approach we can take to address this is to start persisting the Workflow Preference on the NotificationTemplateEntity and modify the `GetPreference` use-case to retrieve the Workflow Preference from the NotificationTemplateEntity during preference evaluation, rather than persist/retrieve the Workflow Preference from the Preference entity.

Reply:
Thanks for bringing this up. Another solution, as I mentioned in our sync, is to reuse the same logic we have for controls. 
Currently, [we store the controls](https://github.com/novuhq/novu/blob/nv-4290-get-v1-controls-step-type-schema/apps/worker/src/app/workflow/usecases/subscriber-job-bound/subscriber-job-bound.usecase.ts#L178) (received from the trigger payload) in the job entity. Since the unsynced flow is not production-related, the storage cost should be negligible. This way, we can use the stored data as needed without modifying the Notification entity for local debugging. 
When we have a better architecture in place, we can create a cleaner solution. 
Let me know your thoughts on this.

Reply:
George is right, after checking the code it needs to start rom the job entity, and then possible evaluate of we have to propagate them to the notifications entity. We need to apply the "copy from blueprint" on even trigger logic for all the workflow definition parts. 

Controls and preferences are part of the workflow definition and we need to treat them as such in the code. Having said that, we can for now make them follow the same logic and in future work, teach the code to consider them one think, that is:

Workflow definition: Steps (the JSON from the TS code that is now the NotificationTemplate), Controls, Preferences, Tags.

Reply:
I'm inclined to go in this direction of storing the `preferences` on the job, that way we don't need to modify the NotificationTemplate & Notification entity and the `preferences` are passed purely as an implementation detail. We won't be locked in to any DAL changes, and the refactoring later will be easier.

Great suggestion!